src/libs/actions/Report.js

```diff
@@ -54,6 +56,35 @@ Onyx.connect({
     },
 });
 
+/**
+ * Map of the most recent non-loading report history items we have stored for a report. Not to be confused with the
+ * reportMaxSequenceNumbers. We can compare these values to those to determine whether there is "available history"
+ * to fetch from the server. maxSequenceNumber is retrieved when getting the reportList whereas these keep track of
+ * the actual reportActions saved in Onyx.
+ */
+const maxReportActionsSequenceNumbers = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+    callback: (val, key) => {
+        if (!key || !val) {
+            return;
+        }
+
+        const reportID = CollectionUtils.extractCollectionItemID(key);
+        const mostRecentAction = _.chain(val)
+            .filter(action => !action.loading)
+            .sortBy('sequenceNumber')
+            .last()
+            .value();
```

Comment:
Oh that's nice. However, I think this is a [lodash](https://lodash.com/docs/4.17.15#maxBy) method and unsure if I can use it with underscore's `chain()`.

Maybe I can use `find()` instead here if we are worried about complexity. The object of report actions _should_ be sorted by default since it's not an array. 

So if we reverse it then take the first one that isn't loading we could achieve something similar. I guess I didn't see much performance impacts from this so I didn't worry about it too much. But that might be a cleaner way to do it.

Reply:
Tried something else here...

- Turned the actions into an array
- Used [`findLastIndex`](https://underscorejs.org/#findLastIndex) to iterate in reverse and get the first non loading

Reply:
> However, I think this is a [lodash](https://lodash.com/docs/4.17.15#maxBy) method and unsure if I can use it with underscore's `chain()`.

Oh derp, didn't even realize there was a distinction 😅, good job with your new solution though!